### ISSUE TITLE
Trying to install GenomeInfoDB, GenomeInfoDBData, and Matrix all lead to non-zero exit status

### ISSUE THREAD

**Issue by genferreri** [Bioconductor/GenomeInfoDb#92]
Hi, I have seen this issue being recurrent and tried many options for the last two days but non yielded to correct installation of any of these packages. I used BiocManager as suggested in other issues, also tried to install from local source, nothing seems to be working. This issue started after updating my R version which now I uninstalled and got back to older version (4.1.3). Still did not solve the problem.

When I try to load Biostrings
```
> library(Biostrings)
Loading required package: GenomeInfoDb
Error: package or namespace load failed for ‘GenomeInfoDb’ in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]):
 there is no package called ‘GenomeInfoDbData’
Error: package ‘GenomeInfoDb’ could not be loaded
```
When I try to install GenomeInfoDbData
```
> BiocManager::install("GenomeInfoDb")
'getOption("repos")' replaces Bioconductor standard repositories, see 'help("repositories", package =
"BiocManager")' for details.
Replacement repositories:
    CRAN: https://cran.rstudio.com/
Bioconductor version 3.14 (BiocManager 1.30.20), R 4.1.3 (2022-03-10)
Old packages: 'Matrix'
Update all/some/none? [a/s/n]: 
a

  There is a binary version available but the source version is later:
       binary source needs_compilation
Matrix  1.5-1  1.5-4              TRUE

Do you want to install from sources the package which needs compilation? (Yes/no/cancel) Yes
installing the source package ‘Matrix’

trying URL 'https://cran.rstudio.com/src/contrib/Matrix_1.5-4.tar.gz'
Content type 'application/x-gzip' length 2177404 bytes (2.1 MB)
==================================================
downloaded 2.1 MB


Attaching package: ‘stats’

The following objects are masked from ‘package:stats4’:

    AIC, BIC, coef, confint, logLik, nobs, profile, update, vcov


Attaching package: ‘graphics’

The following object is masked from ‘package:stats4’:

    plot

Loading required package: ggplot2

Attaching package: ‘ggpubr’

The following object is masked from ‘package:plyr’:

    mutate


Attaching package: ‘rstatix’

The following objects are masked from ‘package:plyr’:

    desc, mutate

The following object is masked from ‘package:stats’:

    filter


Attaching package: ‘purrr’

The following object is masked from ‘package:plyr’:

    compact

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.1     ✔ readr     2.1.4
✔ lubridate 1.9.2     ✔ stringr   1.5.0
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::arrange()   masks plyr::arrange()
✖ purrr::compact()   masks plyr::compact()
✖ dplyr::count()     masks plyr::count()
✖ dplyr::desc()      masks rstatix::desc(), plyr::desc()
✖ dplyr::failwith()  masks plyr::failwith()
✖ dplyr::filter()    masks rstatix::filter(), stats::filter()
✖ dplyr::id()        masks plyr::id()
✖ dplyr::lag()       masks stats::lag()
✖ dplyr::mutate()    masks rstatix::mutate(), ggpubr::mutate(), plyr::mutate()
✖ dplyr::rename()    masks plyr::rename()
✖ dplyr::summarise() masks plyr::summarise()
✖ dplyr::summarize() masks plyr::summarize()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Loading required package: viridisLite
Loading required package: permute
Loading required package: lattice
This is vegan 2.6-4
Loading required package: BiocGenerics

Attaching package: ‘BiocGenerics’

The following objects are masked from ‘package:lubridate’:

    intersect, setdiff, union

The following objects are masked from ‘package:dplyr’:

    combine, intersect, setdiff, union

The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs

The following objects are masked from ‘package:base’:

    Filter, Find, Map, Position, Reduce, anyDuplicated, append,
    as.data.frame, basename, cbind, colnames, dirname, do.call,
    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,
    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,
    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,
    tapply, union, unique, unsplit, which.max, which.min

Loading required package: S4Vectors

Attaching package: ‘S4Vectors’

The following objects are masked from ‘package:lubridate’:

    second, second<-

The following objects are masked from ‘package:dplyr’:

    first, rename

The following object is masked from ‘package:tidyr’:

    expand

The following object is masked from ‘package:plyr’:

    rename

The following objects are masked from ‘package:base’:

    I, expand.grid, unname

Loading required package: IRanges

Attaching package: ‘IRanges’

The following object is masked from ‘package:lubridate’:

    %within%

The following objects are masked from ‘package:dplyr’:

    collapse, desc, slice

The following object is masked from ‘package:purrr’:

    reduce

The following object is masked from ‘package:rstatix’:

    desc

The following object is masked from ‘package:plyr’:

    desc

Loading required package: XVector

Attaching package: ‘XVector’

The following object is masked from ‘package:purrr’:

    compact

The following object is masked from ‘package:plyr’:

    compact

Loading required package: GenomeInfoDb
Error: package or namespace load failed for ‘GenomeInfoDb’ in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]):
 there is no package called ‘GenomeInfoDbData’
Error: package ‘GenomeInfoDb’ could not be loaded
Execution halted

The downloaded source packages are in
	‘/private/var/folders/7t/b96ny1js0lb8dp0n1s48m1ph0000gp/T/RtmpDNaILM/downloaded_packages’
Warning messages:
1: package(s) not installed when version(s) same as or greater than current; use `force = TRUE` to re-install:
  'GenomeInfoDb' 
2: In install.packages(update[instlib == l, "Package"], l, repos = repos,  :
  installation of package ‘Matrix’ had non-zero exit status
```
I have tried to install the version that needs compilation as well as the alternative. Nothing seems to solve the problem. I am not well versed in these type of problems, so if I am not being clear or more information is needed, please let me know.
Thanks in advance

**Comment by hpages** [comment]
Note that this type of questions are better asked on the Bioconductor support site (https://support.bioconductor.org). Also please always provide your `sessionInfo()`.

> When I try to install GenomeInfoDbData... etc

But I don't see that you tried to install GenomeInfoDbData, or did you?

**Comment by genferreri** [comment]
Thanks for replying. I have tried in many different ways. Fails every time. It is very frustrating.
Here is the session info in case you come up with an idea of what could be happening.

```
> sessionInfo()
R version 4.1.3 (2022-03-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS 13.2.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] datasets  utils     grDevices graphics  stats     stats4    methods   base     

other attached packages:
 [1] XVector_0.34.0      IRanges_2.28.0      S4Vectors_0.32.4    BiocGenerics_0.40.0
 [5] vegan_2.6-4         lattice_0.20-45     permute_0.9-7       RColorBrewer_1.1-3 
 [9] viridis_0.6.2       viridisLite_0.4.1   lubridate_1.9.2     stringr_1.5.0      
[13] dplyr_1.1.1         readr_2.1.4         tidyverse_2.0.0     tibble_3.2.1       
[17] tidyr_1.3.0         purrr_1.0.1         forcats_1.0.0       rstatix_0.7.2      
[21] ggpubr_0.6.0        ggplot2_3.4.2       plyr_1.8.8         

loaded via a namespace (and not attached):
 [1] splines_4.1.3     carData_3.0-5     yaml_2.3.7        pillar_1.9.0     
 [5] backports_1.4.1   glue_1.6.2        digest_0.6.31     ggsignif_0.6.4   
 [9] colorspace_2.1-0  htmltools_0.5.5   Matrix_1.5-1      pkgconfig_2.0.3  
[13] broom_1.0.4       zlibbioc_1.40.0   scales_1.2.1      tzdb_0.3.0       
[17] timechange_0.2.0  mgcv_1.8-42       generics_0.1.3    farver_2.1.1     
[21] car_3.1-2         withr_2.5.0       cli_3.6.1         crayon_1.5.2     
[25] magrittr_2.0.3    evaluate_0.20     fansi_1.0.4       nlme_3.1-162     
[29] MASS_7.3-58.3     textshaping_0.3.6 tools_4.1.3       hms_1.1.3        
[33] lifecycle_1.0.3   munsell_0.5.0     cluster_2.1.4     compiler_4.1.3   
[37] systemfonts_1.0.4 rlang_1.1.0       grid_4.1.3        RCurl_1.98-1.12  
[41] rstudioapi_0.14   bitops_1.0-7      rmarkdown_2.21    gtable_0.3.3     
[45] abind_1.4-5       R6_2.5.1          gridExtra_2.3     knitr_1.42       
[49] fastmap_1.1.1     utf8_1.2.3        ragg_1.2.5        stringi_1.7.12   
[53] parallel_4.1.3    Rcpp_1.0.10       vctrs_0.6.1       tidyselect_1.2.0 
[57] xfun_0.38        
```

**Comment by hpages** [comment]
The problem is that you are not showing what happens when you try to install **GenomeInfoDbData**, so make sure to do that when you ask on the Bioconductor support site.

**Comment by genferreri** [comment]
What happens is exactly the same thing as with GenomeInfoDb.

```
BiocManager::install("GenomeDbData")
'getOption("repos")' replaces Bioconductor standard repositories, see
'help("repositories", package = "BiocManager")' for details.
Replacement repositories:
    CRAN: https://cran.rstudio.com/
Bioconductor version 3.14 (BiocManager 1.30.20), R 4.1.3 (2022-03-10)
Installing package(s) 'GenomeDbData'
Old packages: 'lattice', 'Matrix'
Update all/some/none? [a/s/n]: 
a

  There are binary versions available but the source versions are later:
         binary source needs_compilation
lattice 0.20-45 0.21-8              TRUE
Matrix    1.5-1  1.5-4              TRUE

Do you want to install from sources the packages which need compilation? (Yes/no/cancel) no
trying URL 'https://cran.rstudio.com/bin/macosx/contrib/4.1/lattice_0.20-45.tgz'
Content type 'application/x-gzip' length 1157293 bytes (1.1 MB)
==================================================
downloaded 1.1 MB

trying URL 'https://cran.rstudio.com/bin/macosx/contrib/4.1/Matrix_1.5-1.tgz'
Content type 'application/x-gzip' length 5247282 bytes (5.0 MB)
==================================================
downloaded 5.0 MB


The downloaded binary packages are in
	/var/folders/7t/b96ny1js0lb8dp0n1s48m1ph0000gp/T//RtmpMI0cmt/downloaded_packages
Warning message:
package ‘GenomeDbData’ is not available for Bioconductor version '3.14'

A version of this package for your version of R might be available elsewhere,
see the ideas at
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages 
```

**Comment by hpages** [comment]
The name of the package is GenomeInfoDbData, not GenomeDbData.

(The error message: "package ‘GenomeDbData’ is not available for Bioconductor version '3.14'" should give you a clue.)

So after all your posts, we still have no evidence that you actually tried to install GenomeInfoDbData.

But again, this is not the right place to troubleshoot this.

**Comment by genferreri** [comment]
You are actually correct. I appreciate your time and patience.
